clippy_dev/src/fmt.rs

```diff
@@ -31,12 +32,29 @@ struct FmtContext {
     verbose: bool,
 }
 
+// the "main" function of cargo dev fmt
 pub fn run(check: bool, verbose: bool) {
     fn try_run(context: &FmtContext) -> Result<bool, CliError> {
         let mut success = true;
 
         let project_root = clippy_project_root();
 
+        // if we added a local rustc repo as path dependency to clippy for rust analyzer, we do NOT want to
+        // format because rustfmt would also format the entire rustc repo as it is a local
+        // dependency
+        let cargo_toml = {
+            let mut p = project_root.clone();
+            p.push("Cargo.toml");
+            p
+        };
+
+        if std::fs::read_to_string(cargo_toml)
```

Comment:
```suggestion
        if fs::read_to_string(project_root.join("Cargo.toml"))
```
We can simplify this using `join`.

Also, I think importing `std::fs` and using `fs::read_to_string` here would be more idiomatic.